dev-python/berkeleydb/berkeleydb-18.1.15.ebuild

```diff
@@ -0,0 +1,70 @@
+# Copyright 2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{11..14} )
+
+inherit db-use distutils-r1 pypi
+
+DESCRIPTION="Python bindings for Oracle Berkeley DB"
+HOMEPAGE="https://www.jcea.es/programacion/pybsddb.htm"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64"
+
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	<sys-libs/db-6.1:=
+	|| (
+		sys-libs/db:5.3
+		sys-libs/db:4.8
+	)
+"
+BDEPEND="
+	test? (
+		dev-python/packaging[${PYTHON_USEDEP}]
+		dev-python/test[${PYTHON_USEDEP}]
+	)
+"
+
+python_prepare_all() {
+	# This list should be kept in sync with setup.py.
+	if [[ -z ${DB_VER} ]]; then
+		for DB_VER in 5.3 4.8; do
+			has_version "sys-libs/db:${DB_VER}" && break
+		done
+	fi
+
+	# Force version.
+	sed -e "s/db_ver = None/db_ver = (${DB_VER%.*}, ${DB_VER#*.})/" \
+		-e "s/dblib = 'db'/dblib = '$(db_libname ${DB_VER})'/" \
+		-i setup3.py || die
+
+	# no build dir
+	sed -e "s|os.chdir('build')|os.chdir('${S}')|" -i test.py || die
+
+	# Set it ourselves in the test phase
+	sed -e "/os.environ\['BERKELEYDB_SO_PATH'\]/d" -i test.py || die
+
+	mv setup3.py setup.py || die
```

Comment:
Do you actually need this? FWICS it just checks Python versions. Not really useful, but doesn't seem problematic either.

Reply:
The build would fail with the `import setup3` in setup.py, so I just decided this was the simplest solution.